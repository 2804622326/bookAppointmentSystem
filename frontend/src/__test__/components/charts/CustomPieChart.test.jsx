import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import CustomPieChart from '../../../components/charts/CustomPieChart.jsx';

jest.mock('recharts', () => ({
  ResponsiveContainer: ({ children }) => <div>{children}</div>,
  PieChart: ({ children }) => <div data-testid='pie-chart'>{children}</div>,
  Pie: ({ children }) => <div>{children}</div>,
  Cell: () => <div data-testid='cell' />,
  Tooltip: () => <div />,
  Legend: () => <div />,
}));

jest.mock('../../../components/hooks/ColorMapping', () => () => ({ A: '#111' }));

test('renders custom pie chart with cells', () => {
  const { asFragment } = render(<CustomPieChart data={[{ name: 'A', value: 10 }]} />);
  expect(screen.getAllByTestId('cell')).toHaveLength(1);
  expect(asFragment()).toMatchSnapshot();
});
