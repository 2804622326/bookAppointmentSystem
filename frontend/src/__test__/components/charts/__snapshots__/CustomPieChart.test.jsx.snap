// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom pie chart with cells 1`] = `
<DocumentFragment>
  <section
    class="mb-5 mt-5"
  >
    <div>
      <div
        data-testid="pie-chart"
      >
        <div>
          <div
            data-testid="cell"
          />
        </div>
        <div />
        <div />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
